### DISCUSSION TITLE
Is there a way to migrate after pulling existing database without loosing data?

### DISCUSSION THREAD

**Discussion by Knorway** [prisma/prisma#12792] [Q&A]
TL;DR: it seems like I must loose data if I want to work with existing database.

I realized this when integrating a cms called directus which is also known as 'database mirroring' tool you can install on existing database. I initialized prisma project and installed directus on top of it. and if I try to migrate on prisma side I see this error:
```
Drift detected: Your database schema is not in sync with your migration history.

The following is a summary of the differences between the expected database schema given your migrations files, and the actual schema of the database.

It should be understood as the set of changes to get from the expected schema to the actual schema.

If you are running this the first time on an existing database, please make sure to read this documentation page:
https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/troubleshooting-development

We need to reset the SQLite database "database.db" at "file:../database.db".
Do you want to continue? All data will be lost. › (y/N)
```

and turns out it has got nothing to do with the cms. I did some test for few case like creating db and some tables and executing `prisma db pull` and `prisma migrate`, I loose data.

does it only happen on development so you would normally reset dev database to get migration file and do `prisma migrate deploy`?
or should I just use prisma only for querying and stuff when it is not initially started as prisma project? (initially reset for the first migration)

is this kind of operation considered forbidden in relational database world?
I have to confess I do not know much about database or how other orm tools work
and it is highly possible it's from my lack of general knowledge about relational database.
thank you for your time and the great product. cheers!

**Comment by janpio** [comment]
Does https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/baselining help?

**Reply by Knorway** [reply]
@janpio  it seems like baselining needs a migration file to skip and even with `prisma migrate dev --name initial-migration --create-only` after `prisma db pull` following this [instruction](https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/add-prisma-migrate-to-a-project#update-the-development-environment) for that, I get the same error. it feels like --create-only option is not working. isn't it for just creating migration file and not applying it?

**Reply by Knorway** [reply]
resolved but not fully understood why this is happening on first migration.
used this command as described [here](https://github.com/prisma/prisma/issues/11555)
```
mkdir -p prisma/migrations/init
npx prisma migrate diff --preview-feature --from-empty --to-schema-datamodel prisma/schema.prisma --script > prisma/migrations/init/migration.sql
npx prisma migrate resolve --applied init
npx prisma migrate dev # it should say everything is in sync
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
